### PR TITLE
ENH: Upgrade PyTorch Lightning to 1.6.4 

### PR DESCRIPTION
**Pull request by kenza-bouzid**

### CODE REVIEW

File: hi-ml/run_requirements.txt [hunk 1]
```diff
@@ -5,7 +5,7 @@ matplotlib>=3.4.3
 opencv-python-headless>=4.5.1.48
 pandas>=1.3.4
 protobuf<=3.20.1
-pytorch-lightning==1.5.5
+pytorch-lightning==1.6.4
```
[the remark this answers is not in the document]
Comment (kenza-bouzid) [reply hi-ml/run_requirements.txt:8]: sure!

File: hi-ml-histopathology/environment.yml [hunk 2]
```diff
@@ -52,7 +52,7 @@ dependencies:
       - pydicom==2.0.0
       - pyflakes==2.2.0
       - PyJWT==1.7.1
-      - rich==5.1.1
+      - rich==12.4.4
```
Comment (ant0nsc) [inline hi-ml-histopathology/environment.yml:55]: This will also resolve #320

File: hi-ml/src/health_ml/utils/logging.py [hunk 3]
```diff
@@ -367,11 +364,11 @@ def log_learning_rate(module: LightningModule, name: str = "learning_rate") -> N
         raise ValueError("Learning rate logging can only be used during training.")
     single_scheduler = not isinstance(schedulers, list)
     if single_scheduler:
-        schedulers = [schedulers]
+        schedulers = [schedulers]  # type: ignore
```
Comment (ant0nsc) [inline hi-ml/src/health_ml/utils/logging.py:367]: what's the complaint here from mypy?

**Review by ant0nsc — APPROVED** [review]
Good job :-)

**Review by mebristo — APPROVED** [review]
LGTM!

**Pull request merged**